lib/cf_deployer/driver/auto_scaling_group.rb

```diff
@@ -62,21 +62,42 @@ def desired_capacity_reached?
         healthy_instance_count >= desired_capacity
       end
 
-      private
+      def healthy_instance_ids
+        instances = auto_scaling_instances.select do |instance|
+          'HEALTHY'.casecmp(instance.health_status) == 0
+        end
+        instances.map(&:id)
+      end
+
+      def in_service_instance_ids
+        elbs = load_balancers
+        return [] if elbs.empty?
+
+        ids = elbs.collect(&:instances)
+                  .collect(&:health)
+                  .to_a
+                  .collect { |elb_healths|
+                     elb_healths.select { |health| health[:state] == 'InService' }
+                                .map { |health| health[:instance].id }
+                  }
+
+        ids.inject(:&)
+      end
 
       def healthy_instance_count
-        begin
-          count = auto_scaling_instances.count do |instance|
-            instance.health_status == 'HEALTHY' && (load_balancers.empty? || instance_in_service?( instance.ec2_instance ))
-          end
-          Log.info "Healthy instance count: #{count}"
-          count
-        rescue => e
-          Log.info "Unable to determine healthy instance count due to error: #{e.message}"
-          -1
+        AWS.memoize do
+          instances = healthy_instance_ids
+          instances &= in_service_instance_ids unless load_balancers.empty?
```

Comment:
> In the case of a blue/green deployment 
> it's a common case for an ELB to have instances from both color stacks

ugh... but wait, the point of blue/green is to have independent stacks?  each stack should have it's own ELBs, right?

it sounds like maybe you're describing the "canary deployment" scenario from a few weeks ago, where you're wanting to add a new smaller ASG to the existing ELB to slowly test out a change.  in that scenario, wouldn't the health of the new ASG be confirmed prior to being added to the ELB, so in that case, its just the simpler ASG health check?